From 7be4c3e3b02b28aa524d08d3e14196fc47837043 Mon Sep 17 00:00:00 2001
From: Debasish Biswas <debasishbsws.dev@gmail.com>
Date: Fri, 17 Jan 2025 19:26:25 +0530
Subject: [PATCH 4/5] fix(build): remove version constraints from
 requirements.in

In the backend metadata_writer, we aim to use our own modules and tools, which are at much higher versions than the current constraints. To support this, we are removing those constraints from the requirements.in file and letting pip-compile handle the requirements.txt generation.

We regenerate the requirements.txt file using the following command: pip-compile --output-file=- --resolver=backtracking requirements.in > requirements.txt

Signed-off-by: Debasish Biswas <debasishbsws.dev@gmail.com>
---
 backend/metadata_writer/requirements.in  |  8 ++---
 backend/metadata_writer/requirements.txt | 39 +++++++++++-------------
 2 files changed, 22 insertions(+), 25 deletions(-)

diff --git a/backend/metadata_writer/requirements.in b/backend/metadata_writer/requirements.in
index 5e0ef214b..431ec84ac 100644
--- a/backend/metadata_writer/requirements.in
+++ b/backend/metadata_writer/requirements.in
@@ -1,4 +1,4 @@
-kubernetes>=8.0.0,<11.0.0
-ml-metadata==1.14.0
-lru-dict>=1.1.7,<2.0.0
-PyYAML>=5.3,<7
+kubernetes
+ml-metadata
+lru-dict
+PyYAML
diff --git a/backend/metadata_writer/requirements.txt b/backend/metadata_writer/requirements.txt
index 6897654ae..a1fe1b9a4 100644
--- a/backend/metadata_writer/requirements.txt
+++ b/backend/metadata_writer/requirements.txt
@@ -1,35 +1,35 @@
 #
-# This file is autogenerated by pip-compile with Python 3.9
+# This file is autogenerated by pip-compile with Python 3.13
 # by the following command:
 #
-#    pip-compile --output-file=- --resolver=backtracking -
+#    pip-compile --output-file=- --resolver=backtracking - requirements.in
 #
-absl-py==1.4.0
-    # via ml-metadata
-attrs==21.4.0
+absl-py==2.1.0
     # via ml-metadata
 cachetools==5.5.0
     # via google-auth
-certifi==2024.8.30
+certifi==2024.12.14
     # via
     #   kubernetes
     #   requests
-charset-normalizer==3.3.2
+charset-normalizer==3.4.1
     # via requests
-google-auth==2.35.0
+durationpy==0.9
+    # via kubernetes
+google-auth==2.37.0
     # via kubernetes
-grpcio==1.66.1
-    # via ml-metadata
 idna==3.10
     # via requests
 kubernetes==31.0.0
-    # via -r -
+    # via -r requirements.in
 lru-dict==1.3.0
-    # via -r -
-ml-metadata==1.14.0
-    # via -r -
+    # via -r requirements.in
+ml-metadata==0.13.1.dev0
+    # via -r requirements.in
 oauthlib==3.2.2
-    # via requests-oauthlib
+    # via
+    #   kubernetes
+    #   requests-oauthlib
 protobuf==3.20.3
     # via ml-metadata
 pyasn1==0.6.1
@@ -42,7 +42,7 @@ python-dateutil==2.9.0.post0
     # via kubernetes
 pyyaml==6.0.2
     # via
-    #   -r -
+    #   -r requirements.in
     #   kubernetes
 requests==2.32.3
     # via
@@ -52,17 +52,14 @@ requests-oauthlib==2.0.0
     # via kubernetes
 rsa==4.9
     # via google-auth
-six==1.16.0
+six==1.17.0
     # via
     #   kubernetes
     #   ml-metadata
     #   python-dateutil
-urllib3==2.2.3
+urllib3==2.3.0
     # via
     #   kubernetes
     #   requests
 websocket-client==1.8.0
     # via kubernetes
-
-# The following packages are considered to be unsafe in a requirements file:
-# setuptools
-- 
2.39.5 (Apple Git-154)

